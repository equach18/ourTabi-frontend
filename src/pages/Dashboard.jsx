import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import useLocalStorage from "../hooks/useLocalStorage";
import UserProfileSidebar from "../components/UserProfileSidebar";
import FriendPanel from "./user_main/FriendPanel";

function Dashboard({ currentUser, trips }) {
  const [seenWelcome, setSeenWelcome] = useLocalStorage("seenWelcome", false);
  const [showWelcome, setShowWelcome] = useState(!seenWelcome);

  useEffect(() => {
    if (!seenWelcome) {
      const timer = setTimeout(() => {
        setSeenWelcome(true);
        setShowWelcome(false);
      }, 3000);

      return () => clearTimeout(timer);
    }
  }, [seenWelcome]);

  return (
    <div className="flex gap-6 p-6">
      <UserProfileSidebar />

      <div className="flex-1">
        {/* Conditional Welcome Message */}
        {showWelcome && (
          <div className="bg-blue-100 border-l-4 border-blue-500 p-4 mb-6">
            <h1 className="text-3xl font-bold text-blue-500">
              Welcome back, {currentUser?.username}!
            </h1>
          </div>
        )}

        {/* Quick Actions */}
        <div className="mt-6">
          <Link
            to="/trips/new"
            className="bg-green-500 text-white px-6 py-3 rounded-lg hover:bg-green-600"
          >
            + Create a New Trip
          </Link>
        </div>

        {/* User's Trips */}
        <div className="mt-10">
          <h2 className="text-2xl font-semibold text-gray-800">Your Trips</h2>
          <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 mt-6">
            {trips.length > 0 ? (
              trips.map((trip) => (
                <Link
                  to={`/trips/${trip.id}`}
                  key={trip.id}
                  className="border p-4 rounded-lg shadow-md hover:shadow-lg"
                >
                  <h3 className="text-xl font-bold">{trip.title}</h3>
                  <p className="text-gray-600">{trip.destination}</p>
                </Link>
              ))
            ) : (
              <p className="text-gray-500">No trips yet. Start planning now!</p>
            )}
          </div>
        </div>
        <div className="hidden lg:block lg:w-1/4">
          <FriendPanel />
        </div>
      </div>
    </div>
  );
}

export default Dashboard;
