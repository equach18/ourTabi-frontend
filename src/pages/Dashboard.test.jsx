import { screen, waitFor } from "@testing-library/react";
import Dashboard from "./Dashboard";
import {
  renderWithAllProviders,
  testUser1,
  testTrip,
} from "../test-helpers/_testCommons";

// mock components within dashboard
vi.mock("../components/user/UserProfileSidebar", () => ({
  default: () => <div data-testid="UserProfileSidebar" />,
}));
vi.mock("../components/friend/FriendPanel", () => ({
  default: () => <div data-testid="FriendPanel" />,
}));
vi.mock("../components/trip/TripCard", () => ({
  default: ({ trip }) => <div data-testid="TripCard">{trip.title}</div>,
}));

// mock hooks used
vi.mock("../hooks/useLocalStorage", () => ({
  default: () => [false, vi.fn()],
}));

describe("Dashboard", () => {
  function renderDashboard(trips = []) {
    return renderWithAllProviders(
      <Dashboard currentUser={testUser1} trips={trips} />
    );
  }

  it("matches snapshot", () => {
    const { asFragment } = renderDashboard();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders welcome message if not seen", async () => {
    renderDashboard();

    expect(screen.getByText(/welcome back, testuser1/i)).toBeInTheDocument();

    await waitFor(
      () =>
        expect(
          screen.queryByText(/welcome back, testuser1/i)
        ).not.toBeInTheDocument(),
      { timeout: 3500 }
    );
  });

  it("renders trips when provided", () => {
    renderDashboard([testTrip]);

    expect(screen.getByTestId("TripCard")).toBeInTheDocument();
    expect(screen.getByText("Test trip")).toBeInTheDocument();
  });

  it("displays msg if no trips", () => {
    renderDashboard([]);

    expect(
      screen.getByText(/no trips yet\. start planning now!/i)
    ).toBeInTheDocument();
  });

  it("renders Create a New Trip button", () => {
    renderDashboard();

    expect(
      screen.getByRole("link", { name: /\+ create a new trip/i })
    ).toBeInTheDocument();
  });

  it("renders sidebar and friend panel", () => {
    renderDashboard();

    expect(screen.getByTestId("UserProfileSidebar")).toBeInTheDocument();
    expect(screen.getByTestId("FriendPanel")).toBeInTheDocument();
  });
});
