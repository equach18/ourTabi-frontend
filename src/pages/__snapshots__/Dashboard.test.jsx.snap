// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-[calc(100vh-4rem)] grid grid-cols-1 lg:grid-cols-[240px_1fr_280px] gap-6 px-6 pb-10 pt-12"
  >
    <div
      class="w-full"
    >
      <div
        data-testid="UserProfileSidebar"
      />
    </div>
    <div
      class="bg-white p-4 rounded-xl shadow h-full overflow-y-auto"
    >
      <div
        class="bg-blue-100 border-l-4 border-blue-500 p-4 mb-4 rounded-lg shadow-sm"
      >
        <h1
          class="text-2xl font-bold text-blue-500 font-heading"
        >
          Welcome back, testUser1!
        </h1>
      </div>
      <div
        class="flex items-center justify-between mb-4 mt-2"
      >
        <h2
          class="text-xl font-semibold text-zinc-700"
        >
          Your Trips
        </h2>
        <a
          class="bg-emerald-600 text-white px-5 py-2 rounded-lg hover:bg-emerald-700 transition"
          data-discover="true"
          href="/trips/new"
        >
          + Create a New Trip
        </a>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="grid grid-cols-1 sm:grid-cols-2 xl:grid-cols-3 gap-4"
        >
          <p
            class="text-gray-500"
          >
            No trips yet. Start planning now!
          </p>
        </div>
      </div>
    </div>
    <div
      class="hidden lg:block w-full h-[80vh] overflow-y-auto"
    >
      <div
        data-testid="FriendPanel"
      />
    </div>
  </div>
</DocumentFragment>
`;
